terraform/openvdc/provider.go

```diff
@@ -0,0 +1,71 @@
+package openvdc
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+	"os/exec"
+	"strings"
+)
+
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{},
+
+		ResourcesMap: map[string]*schema.Resource{
+			"openvdc_instance": OpenVdcInstance(),
+		},
+	}
+
+}
+
+func RunCmd(name string, arg ...string) (*bytes.Buffer, *bytes.Buffer, error) {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+
+	cmd := exec.Command(name, arg...)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+
+	return &stdout, &stderr, err
+}
+
+func ParseJson(cmdStdOut *bytes.Buffer) (string, error) {
+	var parsed map[string]interface{}
+
+	err := json.Unmarshal(cmdStdOut.Bytes(), &parsed)
+	if err != nil {
+		fmt.Println("Uh-oh! You had an error!")
```

Comment:
It's obvious just to return error object. No need for message to stdout.

Reply:
Additionally this function name is a bit misleading, it only returns a hardcoded parameter.

What about renaming it `ParseState()` or something similar?